serializers/src/main/java/io/pravega/schemaregistry/serializers/JsonGenericDeserializer.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.serializers;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.pravega.schemaregistry.client.SchemaRegistryClient;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+class JsonGenericDeserializer extends AbstractDeserializer<WithSchema<JsonNode>> {
+    private final ObjectMapper objectMapper;
+
+    JsonGenericDeserializer(String groupId, SchemaRegistryClient client,
+                            SerializerConfig.Decoder decoder, EncodingCache encodingCache, boolean encodeHeader) {
+        super(groupId, client, null, false, decoder, encodingCache, encodeHeader);
+        this.objectMapper = new ObjectMapper();
+        objectMapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.ANY);
+        objectMapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
```

Comment:
if `PropertyAccessor.ALL` is set to ANY then the below lines are not required.

Reply:
All doesnt cover fields by default is my understanding.. 
All is for all methods (accessors).
Field and Creator (constructors) are not covered by all afaik.

Reply:
https://github.com/FasterXML/jackson-annotations/blob/629886ecdc4828f2dd2482817c7ba8b0cc886f82/src/main/java/com/fasterxml/jackson/annotation/PropertyAccessor.java#L100-L102 this code snippet from the jackson library indicates ANY overrides everything else. Let me know if I misunderstood it.